### DISCUSSION TITLE
Display of description field

### DISCUSSION THREAD

**Discussion by skripteria** [skripteria/wn-snowflake-plugin#16] [General]
Since we now have a decent content preview in our list view the descrition field has become questionable. I have the impression that the content preview does a better job in guiding the site editor than a verbal description possibly could do.

For a cleaner user interface in this respect I consider 3 options:

1. drop the desc field in the list view but leave it in the edit form
2. drop it in both
3. make the display controllable in the settings

**Comment by arvislacis** [comment]
I think that description should be left as it is - if user doesn't want to see it in the list view then he can always use Winter's List setup button to simply hide that column.

I think descriptions are useful, especially for the fields which don't have initial value set by the developer and are empty. Also they may be useful when some fields could potentially have same contents etc. - then with description user could distinguish equal and/or similar fields.

If we are talking about the descriptions then I personally would like that description gets updated after I run  `php artisan snowflake:sync` - currently the new value doesn't get updated...

And there are some issues when, for example, brackets `()` are used in description..., for example, `{{ test_color|sf('color', 'tests', 'Test description (new)') }}` doesn't get saved properly. Anyways, I have found several edge cases when the Snowflake parser doesn't process `'`, `"`, `,` and `()` properly, I will probably create an issue about that.

**Comment by skripteria** [comment]
Thanks a lot, I will definately question my view on this topic.

The intention why the description only gets synced on insert and not on update was following: 

In some cases it might be useful for the editor to change it and use his own words to make it more helpful from his standpoint. When the description was updated on every sync he permanently would loose his changes because they would be overwritten again.

So there are pros and cons for both ways, maybe an additional field "Edior Notes" could solve this conflict, however it would add another UI element that is probably not used much.

Here I tend to follow your suggestion and better make it updatable and read only on the form.

Im happy to try to support the parsing of special characters there, but I am not sure how much support is be achievable (at least achievable with my limited regex skillls) :) .

**Reply by arvislacis** [reply]
"In some cases it might be useful for the editor to change it and use his own words to make it more helpful from his standpoint. When the description was updated on every sync he permanently would loose his changes because they would be overwritten again." - oh, right, I hadn't thought about this... but, anyways, I am thinking more like this value should be added by developer and also should be updated on the sync..., so probably it may be read-only so editors couldn't override that. But, yes, a think to think about...


One day I was thinking about all parsing and Twig filters - in theory, I think, Snowflake functionality can be realized also with native Twig filter, without doing manual regex scanning etc. but in that case all Snowflake keys would be in one place and it couldn't be divided between separate layouts and pages... Although this approach would have many other benefits, for example, there would not be need for running CLI sync command so often, no problems with regex processing and maybe some others...

But that's a thing I need to think about a lot more.

**Reply by skripteria** [reply]
In theory it is possible to use the original Twig parser within Snowflake, but I couldnt figure out how to do it. In the end it was easier to write my own parsing code so I stopped wasting my time with the Twig attempt.

**Reply by arvislacis** [reply]
That's understandable and is absolutely ok. I haven't worked much with Winter Twig filters too but I will try to do some research - if it would be possible to know if Twig filter is triggered from layout or page then I am sure the existing parser could be rewritten to benefit from native features... Anyways, I would also like to maintain backward compatibility so I need to look if all scenarios would be possible. Anyways, also the current method is just find, maybe some regex expressions and/or argument processing in some cases could be improved a bit but overall works just fine. :)

Anyways, at some time in future maybe some kind of caching or parsing in SfPage->onRun() could happen so running CLI command to update SF keys could be avoided as much as possible - but that's also just an idea, nice thing to have maybe later in future.
